packages/um7/package.py

```diff
@@ -28,38 +25,34 @@ class Um7(Package):
 
     maintainers("penguian")
 
-    depends_on("dummygrib", type=("build", "link"))
     depends_on("fcm", type="build")
+    depends_on("dummygrib", type=("build", "link"))
     depends_on("gcom4@access-esm1.5+mpi", type=("build", "link"))
     depends_on("openmpi@4.0.2:4.1.0", type=("build", "run"))
-    depends_on("netcdf-fortran@4.5.1:4.5.2", type=("build", "link"))
+    depends_on("netcdf-fortran@4.5.2", type=("build", "link"))
     depends_on("oasis3-mct@access-esm1.5", type=("build", "link"))
 
-    variant("omp", default=True, description="Use OpenMP")
-    variant("netcdf", default=True, description="NetCDF")
     variant("opt", default="high", description="Optimization level",
             values=("high", "debug"), multi=False)
 
+    phases = ["edit", "build", "install"]
+
 
     def setup_build_environment(self, env):
+        """
+        Set environment variables to their required values.
+        """
         env.prepend_path("PATH", self.spec["fcm"].prefix.bin)
-        env.prepend_path("CPATH", self.spec["gcom4"].prefix.include)
-        env.prepend_path("CPATH", self.spec["netcdf-fortran"].prefix.include)
-        env.prepend_path("CPATH", self.spec["oasis3-mct"].prefix.include)
-        oasis3_mct_includes = []
-        for header in self.spec["oasis3-mct"].headers:
-            include = os.path.dirname(header)
-            if include not in oasis3_mct_includes:
-                oasis3_mct_includes.append(include)
-                env.prepend_path("CPATH", include)
-        env.prepend_path("LIBRARY_PATH", self.spec["dummygrib"].prefix.lib)
+        oasis3_incs = [
+                join_path(self.spec["oasis3-mct"].prefix.include, subdir)
+                for subdir in ["psmile.MPI1", "mct"]]
+        ideps = ["gcom4", "netcdf-fortran"]
+        incs = [self.spec[d].prefix.include for d in ideps] + oasis3_incs
+        for ipath in incs:
+            env.prepend_path("CPATH", ipath)
+        # The gcom4 library does not contain shared objects and
+        # therefore must be statically linked.
         env.prepend_path("LIBRARY_PATH", self.spec["gcom4"].prefix.lib)
-        env.prepend_path("LIBRARY_PATH", self.spec["netcdf-fortran"].prefix.lib)
-        env.prepend_path("LIBRARY_PATH", self.spec["oasis3-mct"].prefix.lib)
-
-
-    def setup_run_environment(self, env):
-        env.prepend_path("LD_LIBRARY_PATH", self.spec["netcdf-fortran"].prefix.lib)
 
 
     def _bld_path(self):
```

Comment:
Can this be a member variable instead of a member function?

Reply:
Done.